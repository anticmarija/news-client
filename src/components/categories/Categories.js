import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { fetchTop5ArticlesPerCategory } from "../../store/actions/categories.actions";
import CategoriesThumbnail from "../categories-thumbnail/CategoriesThumbnail";
import { Link } from "react-router-dom";
import Loader from "../loader/Loader";
import { CategoriesWrapperStyle } from "./Categories.style";
import { MainHeaderStyle } from "../../style/Shared.style";

const Categories = () => {
  const {
    supportedCategories,
    categoryArticles,
    loadingCategories,
  } = useSelector((state) => state.categories);

  let { activeCountry } = useSelector((state) => state.countries);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchTop5ArticlesPerCategory(activeCountry, supportedCategories));
    // eslint-disable-next-line
  }, [activeCountry, supportedCategories]);

  return loadingCategories ? (
    <Loader />
  ) : (
    <CategoriesWrapperStyle>
      <MainHeaderStyle>Categories</MainHeaderStyle>
      {Object.keys(categoryArticles).map((category) => {
        return (
          <div key={category}>
            <Link
              to={`/country/${activeCountry}/categories/${category.toLowerCase()}`}
            >
              <h2 key={category}>{category}</h2>
            </Link>
            {categoryArticles[category].map((article) => (
              <CategoriesThumbnail
                key={`${article.title}_${article.publishedAt}`}
                article={article}
              />
            ))}
          </div>
        );
      })}
    </CategoriesWrapperStyle>
  );
};

export default Categories;
