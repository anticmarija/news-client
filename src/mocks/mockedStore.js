import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
export const initialState = {
  topNews: {},
  countries: {
    supportedCountries: [{ countryId: "gb", countryName: "Gb" }],
  },
  categories: {},
  article: {
    activeArticle: {
      title: "active",
      urlToImage:
        "https://i.dailymail.co.uk/1s/2020/06/17/16/29719992-0-image-a-23_1592408042536.jpg",
      description: "desc",
    },
  },
};

export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};
