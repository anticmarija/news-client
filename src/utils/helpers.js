export const findNewRoute = (pathname, selectedCountry) => {
  const routeArray = pathname.split("/");
  // Country ID is after 'country'
  const countryIdRouteIndex = routeArray.indexOf("country") + 1;
  routeArray[countryIdRouteIndex] = selectedCountry;

  return routeArray.join("/");
};
