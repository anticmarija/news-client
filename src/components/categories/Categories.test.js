import React from "react";
import Categories from "./Categories";
import { mockedStore } from "../../mocks/mockedStore";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import * as helpers from "../../utils/helpers";
import { MemoryRouter } from "react-router-dom";

describe("Test Categories component", () => {
  it("should render correctly", () => {
    const store = mockedStore();

    jest.spyOn(helpers, "findActiveCountryName").mockReturnValue("gb");

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Categories />
        </MemoryRouter>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
