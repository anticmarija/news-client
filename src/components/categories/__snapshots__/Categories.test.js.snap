// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Categories component should render correctly 1`] = `
<div
  class="Categoriesstyle__CategoriesWrapperStyle-sc-5h4rlm-0 eUa-dA-D"
>
  <h1
    class="Sharedstyle__MainHeaderStyle-sc-1a3bnp9-0 chhIjm"
  >
    Top 5 news by categories from 
    gb
  </h1>
  <div
    class="CategoryHighlightsstyle__CategoryWrapperStyle-z5im9c-0 iSexGd"
  >
    <div
      class="CategoryHighlightsstyle__CategoryTitleWrapper-z5im9c-2 jwXjQn"
    >
      <a
        class="CategoryHighlightsstyle__CategoryLinkStyle-z5im9c-1 dKgQH"
        href="/country/gb/categories/business"
      >
        business
      </a>
      <img
        alt="arrow"
        class="CategoryHighlightsstyle__ArrowIconStyle-z5im9c-4 cUoWuE"
        src="arrow.svg"
      />
    </div>
    <div
      class="CategoryHighlightsstyle__CategoryArticlesStyle-z5im9c-3 jyTEzw"
    >
      <div
        class="Thumbnailstyle__ThumbnailWrapperStyle-sc-123obdz-0 lhHvSI"
      >
        <h2
          class="Thumbnailstyle__ThumbnailTitleStyle-sc-123obdz-3 dSnuxn"
        >
          article1
        </h2>
        <div
          class="Thumbnailstyle__ThumbnailImgStyle-sc-123obdz-1 cwXTzT"
        />
        <div
          class="Thumbnailstyle__ThumbnailDescStyle-sc-123obdz-2 CqRvk"
        >
          <div>
            some desc
          </div>
          <span
            class="Thumbnailstyle__ThumbnailLinkStyle-sc-123obdz-4 VvDHO"
          >
            <a
              href="/country/gb/articles/article1"
            >
              More &gt;
            </a>
          </span>
        </div>
      </div>
      <div
        class="Thumbnailstyle__ThumbnailWrapperStyle-sc-123obdz-0 lhHvSI"
      >
        <h2
          class="Thumbnailstyle__ThumbnailTitleStyle-sc-123obdz-3 dSnuxn"
        >
          article2
        </h2>
        <div
          class="Thumbnailstyle__ThumbnailImgStyle-sc-123obdz-1 cwXTzT"
        />
        <div
          class="Thumbnailstyle__ThumbnailDescStyle-sc-123obdz-2 CqRvk"
        >
          <div>
            some desc
          </div>
          <span
            class="Thumbnailstyle__ThumbnailLinkStyle-sc-123obdz-4 VvDHO"
          >
            <a
              href="/country/gb/articles/article2"
            >
              More &gt;
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="CategoryHighlightsstyle__CategoryWrapperStyle-z5im9c-0 iSexGd"
  >
    <div
      class="CategoryHighlightsstyle__CategoryTitleWrapper-z5im9c-2 jwXjQn"
    >
      <a
        class="CategoryHighlightsstyle__CategoryLinkStyle-z5im9c-1 dKgQH"
        href="/country/gb/categories/sports"
      >
        sports
      </a>
      <img
        alt="arrow"
        class="CategoryHighlightsstyle__ArrowIconStyle-z5im9c-4 cUoWuE"
        src="arrow.svg"
      />
    </div>
    <div
      class="CategoryHighlightsstyle__CategoryArticlesStyle-z5im9c-3 jyTEzw"
    >
      <div
        class="Thumbnailstyle__ThumbnailWrapperStyle-sc-123obdz-0 lhHvSI"
      >
        <h2
          class="Thumbnailstyle__ThumbnailTitleStyle-sc-123obdz-3 dSnuxn"
        >
          article3
        </h2>
        <div
          class="Thumbnailstyle__ThumbnailImgStyle-sc-123obdz-1 cwXTzT"
        />
        <div
          class="Thumbnailstyle__ThumbnailDescStyle-sc-123obdz-2 CqRvk"
        >
          <div>
            some desc
          </div>
          <span
            class="Thumbnailstyle__ThumbnailLinkStyle-sc-123obdz-4 VvDHO"
          >
            <a
              href="/country/gb/articles/article3"
            >
              More &gt;
            </a>
          </span>
        </div>
      </div>
      <div
        class="Thumbnailstyle__ThumbnailWrapperStyle-sc-123obdz-0 lhHvSI"
      >
        <h2
          class="Thumbnailstyle__ThumbnailTitleStyle-sc-123obdz-3 dSnuxn"
        >
          article4
        </h2>
        <div
          class="Thumbnailstyle__ThumbnailImgStyle-sc-123obdz-1 cwXTzT"
        />
        <div
          class="Thumbnailstyle__ThumbnailDescStyle-sc-123obdz-2 CqRvk"
        >
          <div>
            some desc
          </div>
          <span
            class="Thumbnailstyle__ThumbnailLinkStyle-sc-123obdz-4 VvDHO"
          >
            <a
              href="/country/gb/articles/article4"
            >
              More &gt;
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
